### PR TITLE
feat: Exporting all types of Messages so they can be used by consumers

### PR DESCRIPTION
**Pull request by BorntraegerMarc**
Fixes https://github.com/firebase/firebase-admin-node/issues/1146

### CODE REVIEW

**Review by hiranya911 — CHANGES_REQUESTED** [review]
This looks good. Just needs some tests.

**Comment by BorntraegerMarc** [comment]
Will do @hiranya911. Could you point me to the right place to add the tests? I was thinking under `./test/integration/typescript`

File: test/unit/typescript.spec.ts
Comment (hiranya911) [inline test/unit/typescript.spec.ts:32]: Let's try to incorporate this test logic to the existing `messaging.spec` file. No need to carve out a new file for this. Easiest would be to list it under the existing test suite for the `send()` method:

```typescript
      it(`should be fulfilled with a message ID given a TokenMessage`, () => {
        const message: TokenMessage = {token: 'test'}
        mockedRequests.push(mockSendRequest());
        return messaging.send(message)
          .should.eventually.equal('projects/projec_id/messages/message_id');
      });

     // Add similar test cases for other message types.
```

Also note that when you add a new test file, you have to also list it in `index.spec.ts`. Otherwise mocha won't run it. But, let's just add this to the existing unit test file.

**Review by hiranya911 — CHANGES_REQUESTED** [review]
Thanks for the test cases @BorntraegerMarc. Just one comment on how to structure the new test logic. We can merge this once that's addressed.

File: test/unit/typescript.spec.ts
Comment (hiranya911) [reply test/unit/typescript.spec.ts:32]: Or alternatively you can have one test case in the suite for `sendAll()`:

```typescript
    it('should be fulfilled when called with different message types', () => {
      const messageIds = [
        'projects/projec_id/messages/1',
        'projects/projec_id/messages/2',
        'projects/projec_id/messages/3',
      ];
      const tokenMessage: TokenMessage = {token: 'test'};
      const topicMessage: TopicMessage = {topic: 'test'};
      const conditionMessage: ConditionMessage = {condition: 'test'};
      const messages: Message[] = [tokenMessage, topicMessage, conditionMessage];
      mockedRequests.push(mockBatchRequest(messageIds));
      return messaging.sendAll(messages)
        .then((response: BatchResponse) => {
          expect(response.successCount).to.equal(3);
          expect(response.failureCount).to.equal(0);
          response.responses.forEach((resp, idx) => {
            expect(resp.success).to.be.true;
            expect(resp.messageId).to.equal(messageIds[idx]);
            expect(resp.error).to.be.undefined;
          });
        });
    });
```

Comment (BorntraegerMarc) [reply test/unit/typescript.spec.ts]: Makes sense. Thx for the guidance!

---

> Also note that when you add a new test file, you have to also list it in index.spec.ts. Otherwise mocha won't run it

FYI mocha _did_ run the new test. I also verified when I added the new file 🙂 

Screenshot 2021-02-04 at 08 43 04

Source: https://github.com/firebase/firebase-admin-node/runs/1820854315

**Comment by BorntraegerMarc** [comment]
Would it be possible to release a new version after this PR is merged?

I'd appreciate it, so we can update the lib in our project.

**Review by hiranya911 — APPROVED** [review]
Thanks. LGTM 👍 

Will be included in the next release (most likely next week).

**Pull request merged**
